src/utils/nan_payload_tag.hpp

```diff
@@ -0,0 +1,123 @@
+//========================================================================================
+// (C) (or copyright) 2020-2022. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+
+#ifndef UTILS_NAN_PAYLOAD_TAG_HPP_
+#define UTILS_NAN_PAYLOAD_TAG_HPP_
```

Comment:
no, there isn't. I thought it might be useful in some cases where sparse variables are allowed to take any floating point value (so you need some sentinel other than e.g. zero to flag that the data hasn't been initialized) and figured out how to do it, but it isn't used anywhere right now. We could take it out of the PR and introduce it later if someone starts to use it.

Reply:
Let's keep it in for now (but I volunteered you to add the doc later, see https://github.com/parthenon-hpc-lab/parthenon/issues/792)